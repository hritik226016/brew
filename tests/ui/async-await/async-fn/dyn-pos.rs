//@ edition:2018

#![feature(async_closure)]

fn foo(x: &dyn async Fn()) {}
//~^ ERROR the trait `AsyncFn` is not yet dyn-compatible

fn main() {}
