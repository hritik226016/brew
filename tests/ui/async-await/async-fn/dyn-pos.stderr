error[E0802]: the trait `AsyncFn` is not yet dyn-compatible
  --> $DIR/dyn-pos.rs:5:12
   |
LL | fn foo(x: &dyn async Fn()) {}
   |            ^^^^^^^^^^^^^^ `AsyncFn` is not yet dyn compatible
   |
help: consider using an opaque type instead
   |
LL | fn foo(x: &impl async Fn()) {}
   |            ~~~~

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0802`.
